src/tbb/misc_ex.cpp

```diff
@@ -335,38 +348,29 @@ int NumberOfProcessorGroups() {
     return ProcessorGroupInfo::NumGroups;
 }
 
-// Offset for the slot reserved for the first external thread
-#define HoleAdjusted(procIdx, grpIdx) (procIdx + (holeIdx <= grpIdx))
-
 int FindProcessorGroupIndex ( int procIdx ) {
-    // In case of oversubscription spread extra workers in a round robin manner
-    int holeIdx;
-    const int numProcs = theProcessorGroups[ProcessorGroupInfo::NumGroups - 1].numProcsRunningTotal;
-    if ( procIdx >= numProcs - 1 ) {
-        holeIdx = INT_MAX;
-        procIdx = (procIdx - numProcs + 1) % numProcs;
+    int current_grp_idx = ProcessorGroupInfo::HoleIndex;
+    if (procIdx > theProcessorGroups[ProcessorGroupInfo::HoleIndex].numProcs-1  && procIdx <= theProcessorGroups[ProcessorGroupInfo::NumGroups - 1].numProcsRunningTotal-1 ) {
```

Comment:
`theProcessorGroups[ProcessorGroupInfo::HoleIndex]` → `theProcessorGroups[current_grp_idx]` would be more clear, since it is used two more times below.